main.py

```diff
@@ -163,9 +167,33 @@ def handle_errors(self, event):
         The event is sent only if the error is related to a request made
         by flow_manager.
         """
-        xid = event.content["message"].header.xid.value
-        error_type = event.content["message"].error_type
-        error_code = event.content["message"].code
+        message = event.content["message"]
+        connection = event.source
+        switch = connection.switch
+
+        error_data = message.data.pack()
+        # Get the packet responsible for the error
+        error_packet = connection.protocol.unpack(error_data)
+
+        if (message.code == BadActionCode01.OFPBAC_BAD_OUT_PORT or
+           message.code == BadActionCode04.OFPBAC_BAD_OUT_PORT):
```

Comment:
Try to think this way: when we implement OpenFlow 1.4 in `python-openflow`, will we have to come back here and also add support for it? Could we try to make this more "future-proof"?